### ISSUE TITLE
Java API doesn't work with jdbc module

### ISSUE THREAD

**Issue by bowbahdoe** [xtdb/crux#1797]
Both this code

```java
        var xtdb = IXtdb.startNode(NodeConfiguration.builder()
                .with("xtdb.jdbc/connection-pool", ModuleConfiguration.builder()
                        .with("dialect", ModuleConfiguration.builder()
                                .with("xtdb/module", "xtdb.jdbc.psql/->dialect")
                                .build()
                        )
                        .with(
                                "db-spec", ModuleConfiguration.builder()
                                        .with("jdbcUrl", JDBC_DATABASE_URL)
                                        .build()
                        ).build()
                )
                .with("xtdb/tx-log", ModuleConfiguration.builder()
                        .with("xtdb/module", "xtdb.jdbc/->tx-log")
                        .with("connection-pool", "xtdb.jdbc/connection-pool")
                        .build()
                )
                .with("xtdb/document-store", ModuleConfiguration.builder()
                        .with("xtdb/module", "xtdb.jdbc/->document-store")
                        .with("connection-pool", "xtdb.jdbc/connection-pool")
                        .build()
                ).build()
        );
```
        
And the equivalent using `Map.of` crash with 
        
```
Caused by: xtdb.IllegalArgumentException: Arg :db-spec = {jdbcUrl=jdbc:postgresql://127.0.0.1:5432/invite?user=postgres&password=postgres} invalid: {"jdbcUrl" "jdbc:postgresql://127.0.0.1:5432/invite?user=postgres&password=postgres"} - failed: map?
```
        
Which means that the module isn't happy with a Java style map, but there really isn't a good way to make a clojure map without janky interop

**Comment by refset** [comment]
Thanks for raising this. Noting that another non-Clojure workaround would be to use a JSON config, but that may not currently work either if the JDBC module is involved, as per https://github.com/xtdb/xtdb/issues/1588

**Comment by philippkueng** [comment]
We've also just ran into this issue and decided to use the conversion trick detailed here https://groups.google.com/g/clojure/c/1NzLnWUtj0Q - given the small configuration map, would it be acceptable to let it convert the passed in `Map<?,?>`  within XTDB?

**Comment by refset** [comment]
Thanks for commenting & mentioning the workaround @philippkueng - we'll review the options and prioritise accordingly.

**Comment by wotbrew** [comment]
Hello @bowbahdoe, this should be fixed in the next release (see #2757)
